GroupManager/Utils/MsalAppBuilder.cs

```diff
@@ -1,38 +1,62 @@
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+***********************************************************************************************/
+
 using Microsoft.Identity.Client;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
 
 namespace GroupManager.Utils
 {
-    public static class MsalAppBuilder
-    {
-        public static IConfidentialClientApplication BuildConfidentialClientApplication()
-        {
-            IConfidentialClientApplication clientapp = ConfidentialClientApplicationBuilder.Create(Globals.ClientId)
-                  .WithClientSecret(Globals.ClientSecret)
-                  .WithRedirectUri(Globals.RedirectUri)
-                  .WithAuthority(new Uri(Globals.Authority))
-                  .Build();
-
-            MSALPerUserMemoryTokenCache userTokenCache = new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache);
-            return clientapp;
-        }
-
-        public static void ClearUserTokenCache()
-        {
-            IConfidentialClientApplication clientapp = ConfidentialClientApplicationBuilder.Create(Globals.ClientId)
-                  .WithClientSecret(Globals.ClientSecret)
-                  .WithRedirectUri(Globals.RedirectUri)
-                  .WithAuthority(new Uri(Globals.Authority))
-                  .Build();
-
-            // We only clear the user's tokens.
-            MSALPerUserMemoryTokenCache userTokenCache = new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache);
-            userTokenCache.Clear();
-        }
-    }
+	public static class MsalAppBuilder
+	{
+		public static IConfidentialClientApplication BuildConfidentialClientApplication()
+		{
+			IConfidentialClientApplication clientapp = ConfidentialClientApplicationBuilder.Create(Globals.ClientId)
+				  .WithClientSecret(Globals.ClientSecret)
+				  .WithRedirectUri(Globals.RedirectUri)
+				  .WithAuthority(new Uri(Globals.Authority))
+				  .Build();
+
+			MSALPerUserMemoryTokenCache userTokenCache = new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache);
+			return clientapp;
+		}
+
+		public static async Task ClearUserTokenCache()
+		{
+			IConfidentialClientApplication clientapp = ConfidentialClientApplicationBuilder.Create(Globals.ClientId)
+				  .WithClientSecret(Globals.ClientSecret)
+				  .WithRedirectUri(Globals.RedirectUri)
+				  .WithAuthority(new Uri(Globals.Authority))
+				  .Build();
+
+			// We only clear the user's tokens.
+			MSALPerUserMemoryTokenCache userTokenCache = new MSALPerUserMemoryTokenCache(clientapp.UserTokenCache);
+			var userAccount = await clientapp.GetAccountAsync(ClaimsPrincipal.Current.GetMsalAccountId());
+
+			userTokenCache.Clear();
+			await clientapp.RemoveAsync(userAccount);
```

Comment:
>await clientapp.RemoveAsync(userAccount); [](start = 3, length = 41)

This needs to be done before userTokenCache.Clear() as per Bogdan?

Reply:
Good call. Fixed